include/singa/neuralnet/neuron_layer.h

```diff
@@ -478,7 +478,9 @@ class CudnnBMLayer : public BMLayer, public CudnnBase {
   void ComputeFeature(int flag, const vector<Layer*>& srclayers) override;
   void ComputeGradient(int flag, const vector<Layer*>& srclayers) override;
  protected:
+#if CUDNN_MAJOR == 4
   cudnnBatchNormMode_t mode_;
+#endif
```

Comment:
put the whole cudnnbmlayer into #if  ... #endif.
you can git commit --amend -C HEAD to avoid introducing a new commit.